.gitignore

```diff
@@ -0,0 +1,2 @@
+node_modules/
+packs/**/*.d.ts
```

Comment:
how will these be published? do we need a npmignore fill?

Reply:
I'll do a npm pack to see what files get imported, these types will be generated on prepublish, I can make them part of the repo but that kinda cluttered up the repo a bit.

I'll take some screenshots and post them here shortly

Reply:
![image](https://user-images.githubusercontent.com/1120926/74478554-e6666380-4ead-11ea-9cc5-6051af86fc17.png)
This is how amp.d.ts file would look and how the types are generated.

This is how the npm module looks like
![image](https://user-images.githubusercontent.com/1120926/74478773-49f09100-4eae-11ea-982b-1aaea17d010e.png)

![image](https://user-images.githubusercontent.com/1120926/74478777-4e1cae80-4eae-11ea-869e-42b0f060f559.png)